src/App.js

```diff
@@ -53,6 +54,32 @@ class App extends Component {
         }
     }
 
+	/**
+	 * This function will take a text input and checks if it is valid json
+	 * @param {String} jsonFile The json file entered by the user
+	 * @returns {Boolean} The determination if the file is valid json
+	 */
+	validateJSON(jsonFile) {
+		//first a quick way to check the size of the file is to check the character length
+		//Let's assume first of all that a JSON input file should be no larger than 1MB, or 1048576 characters
+		if(jsonFile.length > 1048576) {
+			return false;
+		}
+		//Second check if it is a valid JSON with JSON.parse 
+		var validJson;
+		try {
+			validJson = JSON.parse(jsonFile);
+		} catch(e) {
+			return false; 
+		}
+		//finally check if there are more than 1000 items in the json array or 1000 keys in the object
+		if(Object.keys(validJson).length > 1000) {
```

Comment:
Should we be checking if the object's type is array?

Reply:
Are you looking to just check JSON arrays or all JSON objects? This keys function will return the length of it's base level for objects and arrays (We'll have to look at recursively counting later )

but here is what a Object.keys() would look like the on the following items
Object.keys([1,2,3,4]) =["0", "1", "2", "3"]
Object.keys({"Help":"wanted","test":"Object") = ["Help", "test"]

Reply:
Running sql on an object (key:value) would not make much sense, hence I was thinking of limiting the input to an array of objects. However I don't think underlying engine stops use from doing that, just results are bit weird.

Reply:
I think we could add that as another level of checks.